### PR TITLE
Fix defects

### PR DESCRIPTION
**Pull request by vodik**
Fix some new defects that popped up while reworking the pcap_play and SIPpSocket stuff.

### CODE REVIEW

File: src/call.cpp [hunk 1]
```diff
@@ -1260,6 +1260,12 @@ bool call::executeMessage(message *curmsg)
         }
 
         msg_snd = send_scene(msg_index, &send_status, &msgLen);
+        if (!msg_snd) {
+            /* This will hit connect_if_needed, and if it fails, the
+               entire call is deleted... */
+            ERROR("Call failed, cannot continue safely...");
```
Comment (vodik) [inline src/call.cpp]: This is actually also related to #224.

File: src/scenario.cpp [hunk 2]
```diff
@@ -1857,18 +1860,17 @@ int createStringTable(char * inputString, char *** stringList, int *sizeOfList)
     *stringList = NULL;
     *sizeOfList = 0;
 
+    /* FIXME: temporary workaround: needs rewrite */
+    char *p = const_cast<char*>(strchr(inputString, ','));
```
Comment (vodik) [inline src/scenario.cpp]: I have a rewrite of this method (porting it to std::vector) sitting around, I'll try to get it back in. In the meanwhile, this is at least safe - if very hacky.

File: src/scenario.cpp [hunk 3]
```diff
@@ -236,7 +236,7 @@ double get_double(const char *ptr, const char *what)
 /* Return static buffer to xml value, as with xp_get_value().
  * If the value is enclosed in [brackets], it is assumed to be
  * a command-line supplied keyword value (-key). */
```
Comment (wdoekes) [inline src/scenario.cpp]: The comment is wrong. Static buffer?

**Comment by wdoekes** [comment]
LGTM, apart from the stale static-buffer comment.

File: src/scenario.cpp (hunk 3 above)
Comment (vodik) [inline src/scenario.cpp]: Right, didn't notice that.

**Pull request merged**